File: Практическая работа 5/app/src/main/java/ru/myitschool/lab23/MainActivity.java
package ru.myitschool.lab23;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import android.view.View;
import android.widget.TextView;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;


import ru.myitschool.lab23.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private int count = 0;
    private TextView scoreText;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        scoreText = findViewById(R.id.score_text);
        FloatingActionButton fabTopLeft = findViewById(R.id.fab_top_left);
        FloatingActionButton fabTopRight = findViewById(R.id.fab_top_right);
        FloatingActionButton fabBottomLeft = findViewById(R.id.fab_bottom_left);
        FloatingActionButton fabBottomRight = findViewById(R.id.fab_bottom_right);

        View.OnClickListener fabClickListener = view -> {
            count++;
            scoreText.setText(String.valueOf(count));
        };

        fabTopLeft.setOnClickListener(fabClickListener);
        fabTopRight.setOnClickListener(fabClickListener);
        fabBottomLeft.setOnClickListener(fabClickListener);
        fabBottomRight.setOnClickListener(fabClickListener);

        scoreText.setOnClickListener(view -> {
            count = 0;
            scoreText.setText("0");
            Snackbar.make(view, "Сброс счетчика", Snackbar.LENGTH_SHORT).show();
        });
    }
}


/*
class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

    }
}
*/

File: Практическая работа 8/app/src/main/java/ru/myitschool/lab23/MainActivity.kt
package ru.myitschool.lab23

import android.content.ClipData
import android.content.ClipboardManager
import android.os.Bundle
import android.view.View
import android.widget.ArrayAdapter
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

//class MainActivity : AppCompatActivity() {
//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)
//    }
//}

class MainActivity : AppCompatActivity() {

    private lateinit var editTextSideA: EditText
    private lateinit var editTextSideB: EditText
    private lateinit var editTextSideC: EditText
    private lateinit var spinner: Spinner
    private lateinit var textViewSolution: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editTextSideA = findViewById(R.id.side_a)
        editTextSideB = findViewById(R.id.side_b)
        editTextSideC = findViewById(R.id.side_c)
        spinner = findViewById(R.id.spinner)
        textViewSolution = findViewById(R.id.solution)

        val adapter = ArrayAdapter.createFromResource(this,
            R.array.calculation_options, android.R.layout.simple_spinner_item)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner.adapter = adapter

        findViewById<View>(R.id.calculate).setOnClickListener { calculate() }
        textViewSolution.setOnClickListener { copyToClipboard(textViewSolution.text.toString()) }
    }

    private fun calculate() {
        val sideA = editTextSideA.text.toString().toDouble()
        val sideB = editTextSideB.text.toString().toDouble()
        val sideC = editTextSideC.text.toString().toDouble()

        val result = when (spinner.selectedItemPosition) {
            0-> sideA + sideB + sideC
            1-> Math.sqrt(sideA * sideA + sideB * sideB + sideC * sideC)
            2-> 2 * (sideA * sideB + sideB * sideC + sideC * sideA)
            3-> sideA * sideB * sideC
            else -> 0.0
        }

        textViewSolution.text = String.format("%.5f", result)
    }

    private fun copyToClipboard(text: String) {
        val clipboard = getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
        val clip = ClipData.newPlainText("label", text)
        clipboard.setPrimaryClip(clip)
    }
}
